src/realm/util/client_websocket.hpp

```diff
@@ -16,14 +34,14 @@ namespace realm::util::websocket {
 using port_type = sync::port_type;
 
 // TODO figure out what belongs on config and what belongs on endpoint.
-struct EZConfig {
+struct SocketFactoryConfig {
     util::Logger& logger;
     std::mt19937_64& random;
     util::network::Service& service;
```

Comment:
I think the SocketFactory can just own the service since it'll be responsible for the implementation of the event loop. So we shouldn't have to pass it in via the config.